src/components/autocompleter/autocompleter.jsx

```diff
@@ -0,0 +1,106 @@
+import PropTypes from 'prop-types';
+import React, { forwardRef, useEffect, useState } from 'react';
+import useFetch from '@bloodyaugust/use-fetch';
+import Select from '../select/select.jsx';
+import ListOption from '../list-option/list-option.jsx';
+import mockConstants from '../../mocks/mock-constants.js';
+
+const { API_ROOT } = mockConstants;
+
+const Autocompleter = forwardRef(function Autocompleter(props, ref) {
+  const { execute: fetch } = useFetch();
+  const [models, setModels] = useState(props.models);
+
+  useEffect(() => {
+    fetchModels();
+  }, []);
+
+  function fetchModels(searchString) {
+    fetch(apiEndpoint(searchString))
+      .then((results) => {
+        setModels(results.json.results.map(resultInfo => results.json[resultInfo.key][resultInfo.id]));
+      }).catch((error) => {
+        if (error.error && error.error.type !== 'aborted') {
+          throw error;
+        }
+      });
+  }
+
+  function apiEndpoint(searchString) {
+    if (searchString) {
+      return `${API_ROOT}${props.apiEndpoint}?${props.searchParam}=${searchString}`;
+    }
+
+    return `${API_ROOT}${props.apiEndpoint}`;
+  }
+
+  function displayName(modelInfo) {
+    return modelInfo.title || modelInfo.name || modelInfo.full_name || modelInfo.currency;
+  }
+
+  function onKeyUp(event) {
+    fetchModels(event.target.value);
+  }
+
+  const listOptionRefs = models.map(() => React.createRef());
+  const listOptionElements = models.map((modelInfo, index) => {
+    return (
+      <ListOption key={modelInfo.id} ref={listOptionRefs[index]} value={modelInfo}>
+        {displayName(modelInfo)}
+      </ListOption>
+    );
+  });
+
+  return (
+    <Select
+      label={props.label}
+      id={props.id}
+      name={props.name}
+      className={props.className}
+      listOptionRefs={listOptionRefs}
+      displayValueEvaluator={displayName}
+      onChange={props.onChange}
+      onKeyUp={onKeyUp}
+      required={props.required}
+      readOnly={props.readOnly}
+      placeholder={props.placeholder}
+      errorText={props.errorText}
+      value={props.value}
+      ref={ref}
+    >
+      {listOptionElements}
+    </Select>
+  );
+});
+
+Autocompleter.propTypes = {
+  apiEndpoint: PropTypes.string.isRequired,
+  // singleSelect: PropTypes.bool,
```

Comment:
I take it this is part of what spurred the discussion around single-vs-multiselect in retro